lib/src/models/comment_impl.dart

```diff
@@ -139,25 +158,172 @@ class MoreComments extends RedditBase {
   }
 }
 
-class Comment extends CommentRef {
+/// A fully initialized class which represents a single Reddit comment.
+class Comment extends CommentRef
+    with
+        EditableMixin,
+        GildableMixin,
+        InboxToggleableMixin,
+        InboxableMixin,
+        RedditBaseInitializedMixin,
+        ReplyableMixin,
+        ReportableMixin,
+        SaveableMixin,
+        UserContentMixin,
+        VoteableMixin {
   // CommentModeration get mod; // TODO(bkonyi): implement
 
+  /// Has this [Comment] been approved.
+  bool get approved => data['approved'];
+
+  /// When this [Comment] was approved.
+  ///
+  /// Returns `null` if this [Comment] has not been approved.
+  DateTime get approvedAtUtc => (data['approved_at_utc'] == null)
+      ? null
+      : new DateTime.fromMillisecondsSinceEpoch(
+          data['approved_at_utc'].round() * 1000,
+          isUtc: true);
+
+  /// Which Redditor approved this [Comment].
+  ///
+  /// Returns `null` if this [Comment] has not been approved.
+  RedditorRef get approvedBy => (data['approved_by'] == null)
+      ? null
+      : reddit.redditor(data['approved_by']);
+
+  /// Is this [Comment] archived.
+  bool get archived => data['archived'];
+
+  // TODO(bkonyi): update this definition.
+  // RedditorRef get author => reddit.redditor(data['author']);
+
+  /// The author's flair text, if set.
+  ///
+  /// Returns `null` if the author does not have any flair text set.
+  String get authorFlairText => data['author_flair_text'];
+
+  /// When this [Comment] was removed.
+  ///
+  /// Returns `null` if the [Comment] has not been removed.
+  DateTime get bannedAtUtc => (data['banned_at_utc'] == null)
```

Comment:
Wanna have a Bool get banned/removed => (data['banned_at_utc'] != null).

Than replace the null check with a call to the other getter?

Reply:
See above comments. Probably should just use a helper function here.